src/ttruntime.ts

```diff
@@ -108,7 +113,11 @@ function printReport() {
     const violationsCount = Array.from(violations.values())
         .reduce((x, y) => x + y.length, 0);
     console.log(`Found ${violationsCount} violation${violationsCount === 1 ? '' : 's'}.`);
-    // TODO add ts file localization
+    let i = 1;
+    violations.forEach(async (value: Array<string>, key: Location) => {
+        const location = await getOriginalLocation(key, argv.path || '.')
+        console.error(`${i++}. source: ${location ? `${location.path}:${location.line}:${location.column}` : key}, violations: [${value.join('\n')}]`);
```

Comment:
all of these will print in random order & potentially in different orders between runs. not sure if that is intended. 

It might be more readable to just iterate over each Location to get the promise for the original location, then call `Promise.all` on all of them and print the report after.

If you have performance issues, you can start resolving the disk locations as soon as you get a violation with a location you haven't seen before & save that Promise in a separate map, which you will `await` in this function.